app/controllers/users_controller.rb

```diff
@@ -1,3 +1,25 @@
 class UsersController < ApplicationController
-  def new; end
+  def show
+    @user = User.find params[:id]
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save
+      flash[:success] = "Welcome to the Sample App!"
+      redirect_to @user
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit User::USERS_PARAMS
```

Comment:
params có kiểu dữ liệu gì? 
require có tác dụng gì? In ảnh params.require(:user) ra đây giúp anh
permit có tác dụng gì? Thử thêm 1 cặp key-value vào params.require(:user) và dùng permit với nó xem có chuyện gì xảy ra.
Tìm hiểu về strong parameters

Reply:
- params có kiểu dữ liệu hash
- require(:param) đảm bảo param phải xuất hiện trong hash param, trả về lỗi nếu param đó không tồn tại.
- permit trả lại hash param chỉ bao gồm các key được permit.
- StrongParameters là 1 phương pháp để đối phó với lỗi bảo mật khi sử dụng tính năng Mass Assignment.
- Mass Assignment là 1 tính năng cực kì tiện lợi cho phép update 1 model – khi tạo mới hoặc thay đổi – bằng hash của Ruby.
<img width="1360" alt="Screen Shot 2020-08-04 at 16 35 25" src="https://user-images.githubusercontent.com/68633633/89278597-85effc00-d670-11ea-96ef-e9ce8464e05b.png">